### DISCUSSION TITLE
> Task :react-native-iap:compilePlayDebugKotlin FAILED

### DISCUSSION THREAD

**Discussion by amf-paulinho** [hyochan/react-native-iap#1535] [Q&A]
Node: v14.17.5
OS: Linux

"react": "17.0.2",
"react-native": "0.65.1",
"react-native-iap": "^8.0.0-rc.2",

I tried everything I read online about this error, usually, I can solve the problem by reading docs, etc... But I am stuck with this one.

amfpaulo@valhala:~/_src/jjx/jjxapp$ npx react-native run-android
info Running jetifier to migrate libraries to AndroidX. You can disable it using "--no-jetifier" flag.
Jetifier found 1253 file(s) to forward-jetify. Using 24 workers...
info JS server already running.
info Installing the app...
WARNING:: Please remove usages of `jcenter()` Maven repository from your build scripts and migrate your build to other Maven repositories.
This repository is deprecated and it will be shut down in the future.
See http://developer.android.com/r/tools/jcenter-end-of-service for more information.
Currently detected usages in: root project 'jjxapp', project ':app', project ':react-native-background-downloader', ...
WARNING:: The specified Android SDK Build Tools version (29.0.2) is ignored, as it is below the minimum supported version (30.0.2) for Android Gradle Plugin 4.2.1.
Android SDK Build Tools 30.0.2 will be used.
To suppress this warning, remove "buildToolsVersion '29.0.2'" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.

> Task :react-native-iap:compilePlayDebugKotlin FAILED
w: Runtime JAR files in the classpath should have the same version. These files were found in the classpath:
    /home/amfpaulo/.gradle/caches/transforms-3/6786ed219b01aac6d7fac2224360f0ad/transformed/jetified-kotlin-stdlib-jdk8-1.4.10.jar (version 1.4)
    /home/amfpaulo/.gradle/caches/transforms-3/b2d9124a62abce86428525c4b3742152/transformed/jetified-kotlin-stdlib-jdk7-1.5.30.jar (version 1.5)
    /home/amfpaulo/.gradle/caches/transforms-3/71f81373e43be2b0316ac908c145487e/transformed/jetified-kotlin-stdlib-1.5.30.jar (version 1.5)
    /home/amfpaulo/.gradle/caches/transforms-3/790d07a7c986c56e7ba255d29b71e950/transformed/jetified-kotlin-stdlib-common-1.5.30.jar (version 1.5)
w: Some runtime JAR files in the classpath have an incompatible version. Consider removing them from the classpath

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.9/userguide/command_line_interface.html#sec:command_line_warnings
297 actionable tasks: 2 executed, 295 up-to-date
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
e: /home/amfpaulo/_src/jjx/jjxapp/node_modules/react-native-iap/android/src/play/java/com/dooboolab/RNIap/RNIapPackage.kt: (48, 5): 'createJSModules' overrides nothing

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-iap:compilePlayDebugKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s

error Failed to install the app. Make sure you have the Android development environment set up: https://reactnative.dev/docs/environment-setup.
Error: Command failed: ./gradlew app:installDebug -PreactNativeDevServerPort=8081
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
e: /home/amfpaulo/_src/jjx/jjxapp/node_modules/react-native-iap/android/src/play/java/com/dooboolab/RNIap/RNIapPackage.kt: (48, 5): 'createJSModules' overrides nothing

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-iap:compilePlayDebugKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s

    at makeError (/home/amfpaulo/_src/jjx/jjxapp/node_modules/execa/index.js:174:9)
    at /home/amfpaulo/_src/jjx/jjxapp/node_modules/execa/index.js:278:16
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async runOnAllDevices (/home/amfpaulo/_src/jjx/jjxapp/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/runOnAllDevices.js:106:5)
    at async Command.handleAction (/home/amfpaulo/_src/jjx/jjxapp/node_modules/@react-native-community/cli/build/index.js:192:9)
info Run CLI with --verbose flag for more details.

**Comment by ghyath5** [comment]
Any updates here ?

**Reply by pierroo** [reply]
Hi @amf-paulinho @ghyath5 
Did you guys manage to fix this issue?
I can't get rid of it either; on my end this seems related with the use of react-native-pager-view that uses Kotlin v1.4.0 when iap is using 1.5.0

**Reply by amf-paulinho** [reply]
Man, in my case I missed one important detail during the installation . I just uninstall and reinstall it by the book following the documentation

**Reply by pierroo** [reply]
Thank you for your reply; although that is super weird I don't see how a reinstall could fix the Kotlin versioning issue..; :/
Do you happen to remember which **important detail** you missed during the installation?

**Comment by amf-paulinho** [comment]
Man, i don't remember.  But i am almost sure that problem was about manual
adjustments that often times are needed for components and dependencies.  I
am sure this component plus dependencies need some manual code additions to
react native project files.
Only way for you to solve this is reading all instalation guide. And follow
by the book what's there
